'input' helper component no longer accepts content ({{#input}}{{/input}} now crashes)

Upon upgrading from 2.11 to 2.13, the built-in 'input' component no longer accepts content - and that might be reasonable, but it crashes the component with a cryptic console error rather than failing during build or providing a useful error.

I'm not entirely sure why, but I had in a template:
```
{{#input value=myValue focus-out='myAction'}}
   some content
{{/input}}
```

In prior versions (<2.11) this built and ran fine. Upon upgrading to 2.13 however, simply having this pattern on a page crashes the component:
```
{{#input}}{{/input}}
```

 with the error:
> ember.debug.js:19818 Error: input not found, and the catch-all block handler didn't handle it
>     at Object.debugAssert [as assert] (ember.debug.js:1498)
>     at Blocks.compile (ember.debug.js:1202)
>     at ember.debug.js:1201
>     at Compilers.compile (ember.debug.js:1201)
>     at InlineBlock.splat (ember.debug.js:1200)
>     at InlineBlock.compile (ember.debug.js:1200)
>     at VM.invokeBlock (ember.debug.js:1355)
>     at ember.debug.js:1103
>     at AppendOpcodes.evaluate (ember.debug.js:1102)
>     at VM.next (ember.debug.js:1353)
>     at VM.execute (ember.debug.js:1353)
>     at VM.resume (ember.debug.js:1353)
>     at TryOpcode.handleException (ember.debug.js:1355)
>     at UpdatingVMFrame.handleException (ember.debug.js:1356)
>     at UpdatingVM._throw [as throw] (ember.debug.js:1355)
>     at Assert.evaluate (ember.debug.js:1103)
>     at UpdatingVM.execute (ember.debug.js:1355)
>     at RenderResult.rerender (ember.debug.js:1343)
>     at RootState._this.render (ember.debug.js:14014)
>     at Object.runInTransaction (ember.debug.js:25212)
>     at InteractiveRenderer._renderRoots (ember.debug.js:14275)
>     at InteractiveRenderer._renderRootsTransaction (ember.debug.js:14313)
>     at InteractiveRenderer._revalidate (ember.debug.js:14352)
>     at invokeWithOnError (ember.debug.js:2197)
>     at Queue.flush (ember.debug.js:2256)
>     at DeferredActionQueues.flush (ember.debug.js:2380)
>     at Backburner.end (ember.debug.js:2450)
>     at Backburner.run (ember.debug.js:2564)
>     at Object.run (ember.debug.js:24451)
>     at ActionState.handler (ember.debug.js:13785)
>     at HTMLAnchorElement.<anonymous> (ember.debug.js:44352)
>     at HTMLBodyElement.dispatch (jquery.js:5206)
>     at HTMLBodyElement.elemData.handle (jquery.js:5014)


It would be nice if the pattern was caught during build or at least some notification somewhere that the pattern was illegal.

--- a/packages/ember-template-compiler/lib/plugins/assert-input-helper-without-block.js
+++ b/packages/ember-template-compiler/lib/plugins/assert-input-helper-without-block.js
@@ -1,0 +1,24 @@
+import { assert } from 'ember-debug';
+import calculateLocationDisplay from '../system/calculate-location-display';
+
+export default function errorOnInputWithContent(env) {
+  let { moduleName } = env.meta;
+
+  return {
+    name: 'assert-input-helper-without-block',
+
+    visitors: {
+      BlockStatement(node) {
+        if (node.path.original !== 'input') { return; }
+
+        assert(assertMessage(moduleName, node));
+      }
+    }
+  };
+}
+
+function assertMessage(moduleName, node) {
+  let sourceInformation = calculateLocationDisplay(moduleName, node.loc);
+
+  return `The {{input}} helper cannot be used in block form. ${sourceInformation}`;
+}

--- a/packages/ember-template-compiler/lib/plugins/index.js
+++ b/packages/ember-template-compiler/lib/plugins/index.js
@@ -15,6 +15,7 @@ import TransformEachInIntoEach from './transform-each-in-into-each';
 import TransformHasBlockSyntax from './transform-has-block-syntax';
 import TransformDotComponentInvocation from './transform-dot-component-invocation';
 import ExtractPragmaTag from './extract-pragma-tag';
+import AssertInputHelperWithoutBlock from './assert-input-helper-without-block';
 import {
   GLIMMER_CUSTOM_COMPONENT_MANAGER
 } from 'ember/features';
@@ -35,7 +36,8 @@ const transforms = [
   TransformInputTypeSyntax,
   TransformAttrsIntoArgs,
   TransformEachInIntoEach,
-  TransformHasBlockSyntax
+  TransformHasBlockSyntax,
+  AssertInputHelperWithoutBlock
 ];
 
 if (GLIMMER_CUSTOM_COMPONENT_MANAGER) {

--- a/packages/ember-template-compiler/tests/plugins/assert-input-helper-without-block-test.js
+++ b/packages/ember-template-compiler/tests/plugins/assert-input-helper-without-block-test.js
@@ -1,0 +1,16 @@
+import { compile } from '../../index';
+
+QUnit.module('ember-template-compiler: assert-input-helper-without-block');
+
+QUnit.test('Using {{#input}}{{/input}} is not valid', function() {
+  expect(1);
+
+  let expectedMessage =
+    `The {{input}} helper cannot be used in block form. ('baz/foo-bar' @ L1:C0) `;
+
+  expectAssertion(() => {
+    compile('{{#input value="123"}}Completely invalid{{/input}}', {
+      moduleName: 'baz/foo-bar'
+    });
+  }, expectedMessage);
+});

